### ISSUE TITLE
add multi thread support for ATMS converter 

### ISSUE THREAD

**Issue by BenjaminRuston** [JCSDA-internal/ioda-converters#783]
## Description
The ATMS converter from raw HDF5 or netCDF4 files should have multi threading support added

## Requirements
python3 and h5py libraries

## Acceptance Criteria (Definition of Done)
successfully produce a IODA file both for the ctest case and for a series of files

**Comment by srherbener** [comment]
Here are some sample input files on Orion:
```
/work/noaa/da/bruston/data_repos/atms/20220217/{SNDR.J1.ATMS.20220217T0000.m06.g001.L1B.std.v03_15.G.220217065428.nc4,SNDR.J1.ATMS.20220217T0006.m06.g002.L1B.std.v03_15.G.220217065438.nc4}
```

And this is the feature branch with the code: `feature/atms_multi_threaded_converter`

**Comment by srherbener** [comment]
@BenjaminRuston I found and fixed an issue with the `concat_obs_data` function that was causing issues with any variables that were 2D (or higher rank). I pushed these changes to github.

That fixed a problem with data getting scrambled, but did not fix the problem in the large file with garbage in the `ObsValue/brightness_temperature` variable. That issue still exists, and I think it is related to the chunk size limit we put on the variables. It looks like the Python API won't write a partial chunk. The limit is set to 10000 on nlocs and the first 20000 values look good and the remain 5920 values look corrupt. The obs_data dictionary being handed off to the writer.BuildIoda() function looked good.

I'm just logging these notes here if we decide to put this in the icebox, and we don't look at this for a while.
